modern/src/settings/ComputedAttributePage.js

```diff
@@ -0,0 +1,92 @@
+import React, { useState } from 'react';
+import { Accordion, AccordionSummary, AccordionDetails, makeStyles, Typography,  FormControl, InputLabel, MenuItem, Select, TextField } from "@material-ui/core";
+import t from '../common/localization';
+import EditItemView from '../EditItemView';
+
+import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
+import positionAttributes from '../attributes/positionAttributes';
+
+
+const useStyles = makeStyles(() => ({
+  details: {
+    flexDirection: 'column',
+  },
+}));
+
+const ComputedAttributePage =() => {
+
+  const classes = useStyles();
+  const [item, setItem] = useState();
+  const [key, setKey] = useState();
+
+  const options = Object.entries(positionAttributes).map(([key, value]) => ({
+    key,
+    name: value.name,
+    type: value.type,
+  }));
+
+  return (
+    <EditItemView endpoint="/attributes/computed" item={item} setItem={setItem}>
+      {item &&
+        <Accordion defaultExpanded>
+          <AccordionSummary expandIcon={<ExpandMoreIcon />}>
+            <Typography variant="subtitle1">
+              {t('sharedRequired')}
+            </Typography>
+          </AccordionSummary>
+          <AccordionDetails className={classes.details}>
+            <TextField
+              margin="normal"
+              value={item.description || ''}
+              onChange={event => setItem({...item, description: event.target.value})}
+              label={t('sharedDescription')}
+              variant="filled" />
+            <FormControl variant="filled" margin="normal" fullWidth>
+              <InputLabel>{t('sharedAttribute')}</InputLabel>
+              <Select 
+                value={item.attribute || ''} 
+                onChange={(e) => {
+                  const newValue = e.target.value;
+                  const positionAttribute = positionAttributes[newValue];
+                  setKey(newValue);
+                  if(positionAttribute && positionAttribute.type) {
```

Comment:
Please fix formatting. Also I would recommend extracting this into a separate method. Too much code for inline method.